### PR TITLE
Consolidate proto-compiler logic into main.rs

### PR DESCRIPTION
**Pull request by romac**
Closes: #241

## Description

This PR consolidates the proto-compiler crate into a single binary.

Cloning the SDK, compiling the .proto files to Rust, and copying them to the appropriate location now only happen at runtime, whereas it was previously split between build time (via `build.rs`) and runtime (via `main.rs`).

This should let us improve on the UX of the proto-compiler binary in the future, eg. by introducing a proper CLI, etc.

______

For contributor use:

- [ ] Unit tests written
- [ ] Added test to CI if applicable 
- [ ] Updated CHANGELOG_PENDING.md
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by ancazamfir** [comment]
Would it be possible to add a README.md in `proto-compiler` that describes the steps to take for a new version to be generated up to the `proto` crate publishing? Also will the compiled files still be dropped in the `proto` crate?

**Comment by romac** [comment]
> Would it be possible to add a README.md in proto-compiler

Done.

> Also will the compiled files still be dropped in the proto crate?

Yes!

**Review by ancazamfir — APPROVED** [review]

File: proto-compiler/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# ibc-proto-compiler
+
+The `ibc-proto-compiler` is a simple command-line tool to automate the compilation of [Protocol Buffers](https://developers.google.com/protocol-buffers) message definitions from the [Cosmos SDK](https://github.com/cosmos/cosmos-sdk) to Rust source code with [Prost](https://lib.rs/crates/prost), for use in the [`ibc-proto` crate](https://lib.rs/crates/ibc-proto) in the [`ibc-rs` project](https://github.com/informalsystems/ibc-rs/).
+
+## Usage
+
+From within the `ibc-proto-compiler` directory, run the following command to clone the Cosmos SDK repository, generate the Rust sources from the Protobuf definitions, and copy them to the `ibc-proto` crate within the `ibc-rs` project:
```
Comment (greg-szabo) [inline proto-compiler/README.md:7]:
```suggestion
From within the `proto-compiler` directory, run the following command to clone the Cosmos SDK repository, generate the Rust sources from the Protobuf definitions, and copy them to the `ibc-proto` crate within the `ibc-rs` project:
```

Comment (greg-szabo) [reply proto-compiler/README.md:7]:
```suggestion
From within the `ibc-proto-compiler` crate, run the following command to clone the Cosmos SDK repository, generate the Rust sources from the Protobuf definitions, and copy them to the `ibc-proto` crate within the `ibc-rs` project:
```

Comment (greg-szabo) [reply proto-compiler/README.md:7]: one or the other is more correct

File: proto-compiler/src/main.rs [hunk 2]
```diff
@@ -1,36 +1,129 @@
...
+use std::path::{Path, PathBuf};
+
+use git2::Repository;
+use tempdir::TempDir;
 use walkdir::WalkDir;
 
-pub(crate) fn main() {
-    let ibc_proto_path = "../proto/src/prost";
+fn main() {
+    let root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
```
Comment (greg-szabo) [inline proto-compiler/src/main.rs:11]: nice!

File: proto-compiler/src/main.rs [hunk 3]
```diff
@@ -1,36 +1,129 @@
...
+fn clone_cosmos_sdk() -> PathBuf {
+    let sdk_dir = var("SDK_DIR").unwrap_or_else(|_| "target/cosmos-sdk".to_string());
+
+    if Path::new(&sdk_dir).exists() {
+        println!("[info ] Found Cosmos SDK source at '{}'", sdk_dir);
+    } else {
+        println!("[info ] Cloning cosmos/cosmos-sdk repository...");
+
+        let url = "https://github.com/cosmos/cosmos-sdk";
+        Repository::clone(url, &sdk_dir).unwrap();
```
Comment (greg-szabo) [inline proto-compiler/src/main.rs:32]: I'll have a more robust solution to this in the tendermint-proto crate. Later, for a next update.

**Review by greg-szabo — APPROVED** [review]
I added one minor documentation clarification annoyment but I feel this is good to go.

**Pull request merged**